Error occurs when running tests after introducing test framework

After introducing the test framework, errors are occurring when running tests.

  Steps to reproduce:
  1. Run uv run --group test pytest tests/test_api.py
  2. Observe test execution errors

  @claude Please investigate and fix the test errors that are occurring after the test framework
  introduction.

Detail: 
ImportError while loading conftest 'starting-ragchatbot-codebase/tests/conftest.py'.
tests/conftest.py:16: in <module>
    from app import app
backend/app.py:119: in <module>
    app.mount("/", StaticFiles(directory="../frontend", html=True), name="static")
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/starlette/staticfiles.py:56: in __init__
    raise RuntimeError(f"Directory '{directory}' does not exist")
E   RuntimeError: Directory '../frontend' does not exist

--- a/backend/app.py
+++ b/backend/app.py
@@ -116,4 +116,7 @@ class DevStaticFiles(StaticFiles):
     
     
 # Serve static files for the frontend
-app.mount("/", StaticFiles(directory="../frontend", html=True), name="static")
+# Use absolute path and make it optional during testing
+frontend_path = Path(__file__).parent.parent / "frontend"
+if frontend_path.exists():
+    app.mount("/", StaticFiles(directory=str(frontend_path), html=True), name="static")

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -27,7 +27,8 @@ def client():
 @pytest.fixture
 async def async_client():
     """Create an async test client for the FastAPI app."""
-    async with AsyncClient(app=app, base_url="http://test") as ac:
+    from httpx import ASGITransport
+    async with AsyncClient(transport=ASGITransport(app=app), base_url="http://test") as ac:
         yield ac
 
 

